operate/wallet/wallet_recovery_manager.py

```diff
@@ -0,0 +1,210 @@
+            )
+
+        if not password:
+            raise ValueError("'password' must be a non-empty string.")
+
+        if not bundle_id:
+            raise ValueError("'bundle_id' must be a non-empty string.")
+
+        root = self.path.parent  # .operate root
+        wallets_path = root / WALLETS_DIR
+        new_root = self.path / bundle_id / RECOVERY_NEW_OBJECTS_DIR
+        new_wallets_path = new_root / WALLETS_DIR
+        old_root = self.path / bundle_id / RECOVERY_OLD_OBJECTS_DIR
+
+        if not new_root.exists() or not new_root.is_dir():
+            raise ValueError(f"Recovery bundle {bundle_id} does not exist.")
+
+        if old_root.exists() and old_root.is_dir():
+            raise ValueError(f"Recovery bundle {bundle_id} has been executed already.")
+
+        new_user_account = UserAccount.load(new_root / USER_JSON)
+        if not new_user_account.is_valid(password=password):
+            raise ValueError("Password is not valid.")
+
+        new_wallet_manager = MasterWalletManager(
+            path=new_wallets_path, logger=self.logger, password=password
+        )
+
+        ledger_types = {item.ledger_type for item in self.wallet_manager}
+        new_ledger_types = {item.ledger_type for item in new_wallet_manager}
+
+        if ledger_types != new_ledger_types:
+            raise WalletRecoveryError(
+                f"Ledger type mismatch: {ledger_types=}, {new_ledger_types=}."
+            )
+
+        for wallet in self.wallet_manager:
+            new_wallet = next(
+                (w for w in new_wallet_manager if w.ledger_type == wallet.ledger_type)
+            )
+
+            all_backup_owners = set()
+            for chain, safe in wallet.safes.items():
+                ledger_api = wallet.ledger_api(chain=chain)
+                owners = get_owners(ledger_api=ledger_api, safe=safe)
+                if new_wallet.address not in owners:
+                    raise WalletRecoveryError(
+                        f"Incorrect owners. Wallet {new_wallet.address} is not an owner of Safe {safe} on {chain}."
+                    )
+                if wallet.address in owners:
+                    _report_issue(
+                        f"Inconsistent owners. Current wallet {wallet.address} is still an owner of Safe {safe} on {chain}."
+                    )
+                if len(owners) != 2:
+                    _report_issue(
+                        f"Inconsistent owners. Safe {safe} on {chain} has {len(owners)} != 2 owners."
+                    )
+                all_backup_owners.update(set(owners) - {new_wallet.address})
+
+            if len(all_backup_owners) != 1:
+                _report_issue(
+                    f"Inconsistent owners. Backup owners differ across Safes on chains {', '.join(chain.value for chain in wallet.safes.keys())}. "
+                    f"Found backup owners: {', '.join(map(str, all_backup_owners))}."
+                )
```

Comment:
it would be better to have this check in `/initiate`, wdyt?

Reply:
This check should be done here for sure. Being done on the initiate as well is optional. I would go for not including it there to avoid cluttering the endpoint.

So the "initiate" always create an "inocuous" temporary object.